### DISCUSSION TITLE
renovate and NPM 7 when lockFileMaintenance: { enabled: true }

### DISCUSSION THREAD

**Discussion by rvillane** [renovatebot/renovate#8638] [Request Help]
Hi, a few days ago NPM 7 went GA (see https://github.blog/2021-02-02-npm-7-is-now-generally-available/) and among other changes, it introduces a new package-lock format and wondering if there is any implications for renovate when lockFileMaintenance: is enabled and:

- existing package-lock.json is originally generated using NPM 6 ?
- existing package-lock.json is originally generated using NPM 7 ?

thanks

**Comment by viceice** [comment]
It should work, if no engine is defined for npm renovate will check the lockfile version to decide for npm version to use.
